examples/seismic/viscoacoustic/operators.py

```diff
@@ -0,0 +1,208 @@
+
+    return [u_v, u_r, u_p]
+
+def viscoacoustic_ren(v, p, vp, irho, rho, qp, f0, s, damp, **kwargs):
+    """
+    Stencil created from Ren et al. (2014) viscoacoustic wave equation 
+
+    Parameters
+    ----------
+    v : VectorTimeFunction
+        The computed particle velocity.
+    p : TimeFunction
+        The computed solution.
+    vp : Function or float
+        The time-constant velocity. 
+    irho : Function
+        The time-constant inverse density.
+    rho : Function
+        The time-constant density.
+    qp : Function
+        The P-wave quality factor.
+    f0: float or Scalar
+        The dominant frequency
+    s : float or Scalar
+        The time dimension spacing.
+    damp : Function
+        The damping field for absorbing boundary condition.
+    """
+    # Angular frequency 
+    w = 2. * np.pi * f0
+
+    # Define PDE
+    pde_v = v - s * irho * grad(p)
+
+    u_v = Eq(v.forward, damp * pde_v)
+
+    pde_u = p - s * vp * vp * rho * div(v.forward) + s * ((vp * vp * rho) / \
+            (w * qp)) * div(irho * grad(p))
+
+    u_p = Eq(p.forward, damp * pde_u)
+
+    return [u_v, u_p]
+
+def viscoacoustic_deng_mcmechan(v, p, vp, irho, rho, qp, f0, s, damp, **kwargs):
+    """
+    Stencil created from Deng and McMechan (2007) viscoacoustic wave equation
+
+    Parameters
+    ----------
+    v : VectorTimeFunction
+        The computed particle velocity.   
+    p : TimeFunction
+        The computed solution.
+    vp : Function or float
+        The time-constant velocity.
+    irho : Function
+        The time-constant inverse density.
+    rho : Function
+        The time-constant density.
+    qp : Function
+        The P-wave quality factor.
+    f0: float or Scalar
+        The dominant frequency
+    s : float or Scalar
+        The time dimension spacing.
+    damp : Function
+        The damping field for absorbing boundary condition.
+    """
+    # Angular frequency 
+    w = 2. * np.pi * f0
+    # Define PDE
+    pde_v = v - s * irho * grad(p)
+    
+    u_v = Eq(v.forward, damp * pde_v)
+    
+    pde_p = p - s * vp * vp * rho * div(v.forward) - s * (w / qp) * p
+    
+    u_p = Eq(p.forward, damp * pde_p)
+
+    return [u_v, u_p]
+
+def ForwardOperator(model, geometry, space_order=4, equation=1, save=False, **kwargs):
+    """
+    Construct method for the forward modelling operator in an viscoacoustic media.
+
+    Parameters
+    ----------
+    model : Model
+        Object containing the physical parameters.
+    geometry : AcquisitionGeometry
+        Geometry object that contains the source (SparseTimeFunction) and
+        receivers (SparseTimeFunction) and their position.
+    space_order : int, optional
+        Space discretization order.
+    save : int or Buffer
+        Saving flag, True saves all time steps, False saves three buffered
+        indices (last three time steps). Defaults to False.
+    equation : selects a visco-acoustic equation from the options below:
```

Comment:
Would call it `kernel` like in acoustic or tti and use explicit names rather than numbers then you can make a kernel map like in tti.